statsmodels/base/distributed_estimation.py

```diff
@@ -0,0 +1,400 @@
+    estimate theta hat.
+
+    Parameters
+    ----------
+    X_beta : array-like
+        The weighted design matrix for the current machine
+    pi : scalar
+        Index of the current variable
+    p : scalar
+        Number of variables
+    n : scalar
+        sample size of current machine
+    alpha : scalar or array-like
+        The penalty weight.  If a scalar, the same penalty weight
+        applies to all variables in the model.  If a vector, it
+        must have the same length as `params`, and contains a
+        penalty weight for each coefficient.
+
+    Returns
+    -------
+    An array-like object of length p-1
+
+    Notes
+    -----
+
+    gamma_hat_i = arg min 1/(2n) ||X_beta,i - X_beta,-i gamma||_2^2
+                          + alpha ||gamma||_1
+    """
+
+    # TODO this is not a very elegant solution, we can't concat
+    # range in Py 3 because it is now a generator.
+    ind = list(range(pi)) + list(range(pi + 1, p))
+
+    # TODO use elastic net optimization routine directly
+    tmod = OLS(X_beta[:, pi], X_beta[:, ind])
+
+    # TODO this is currently going to fail with some alpha
+    # values, the issue is that may have an array for of
+    # alphas for beta hat and we want each individual element
+    # here
+    gamma_hat = tmod.fit_regularized(alpha=alpha).params
+
+    return gamma_hat
+
+
+def _gen_tau_hat(X_beta, gamma_hat, pi, p, n, alpha):
+    """generates the tau hat value for the pith variable, used to
+    estimate theta hat.
+
+    Parameters
+    ----------
+    X_beta : array-like
+        The weighted design matrix for the current machine
+    gamma_hat : array-like
+        The gamma_hat values for the current variable
+    pi : scalar
+        Index of the current variable
+    p : scalar
+        Number of variables
+    n : scalar
+        sample size of current machine
+    alpha : scalar or array-like
+        The penalty weight.  If a scalar, the same penalty weight
+        applies to all variables in the model.  If a vector, it
+        must have the same length as `params`, and contains a
+        penalty weight for each coefficient.
+
+    Returns
+    -------
+    A scalar
+
+    Notes
+    -----
+
+    tau_hat_i = sqrt(1/n ||X_beta,i - X_beta,-i gamma_hat||_2^2
+                     + alpha ||gamma||_1)
+    """
+
+    ind = list(range(pi)) + list(range(pi + 1, p))
+    d = np.linalg.norm(X_beta[:, pi] - X_beta[:, ind].dot(gamma_hat))**2
+    d = np.sqrt(d / n + alpha * np.linalg.norm(gamma_hat, 1))
+    return d
+
+
+def _gen_theta_hat(gamma_hat_l, tau_hat_l, p):
+    """generates the theta hat matrix
+
+    Parameters
+    ----------
+    gamma_hat_l : list
+        A list of array-like object where each object corresponds to
+        the gamma hat values for the corresponding variable, should
+        be length p.
+    tau_hat_l : list
+        A list of scalars where each scalar corresponds to the tau hat
+        value for the corresponding variable, should be length p.
```

Comment:
those are pretty tautological descriptions if we don't know what the greek letter stands for.

A glossary for greek translation would be useful until those have descriptive names.